bilby/core/prior/dict.py

```diff
@@ -519,6 +603,9 @@ def prob(self, sample, **kwargs):
         return self.check_prob(sample, prob)
 
     def check_prob(self, sample, prob):
+        if self.conversion_function == self.default_conversion_function:
```

Comment:
I wanted to check if the normalization factor integration is necessary by checking if the prior is constrained because if not the new error estimation will fail. However, this should be handled better on the level of the error estimation anyway because it will also fail at the moment of the constraint does not affect the list of keys selected.

Are there situations where checking if any prior in the PriorDict is a constraint and only applying the normalization if so would be unsafe?